gin/v1/middleware.go

```diff
@@ -0,0 +1,37 @@
+package gin
+
+import (
+	"log"
+	"net/http/httptest"
+	"strings"
+
+	"github.com/betacraft/yaag/middleware"
+	"github.com/betacraft/yaag/yaag"
+	"github.com/betacraft/yaag/yaag/models"
+	"gopkg.in/gin-gonic/gin.v1"
+)
+
+func Document() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		if !yaag.IsOn() {
+			return
+		}
+		writer := httptest.NewRecorder()
+		apiCall := models.ApiCall{}
+		middleware.Before(&apiCall, c.Request)
+		c.Next()
+		if writer.Code != 404 {
```

Comment:
Rather than checking for 404, if we just check for 200 and process the response. Anyways we will need only success responses in the document

Reply:
A lot of devs use proper http codes. Like created, accepted etc. That's why it's checking for not-404. We can actually check if error codes are 2xx. That way only success codes will get stored in doc.